File: exercises/practice/meetup/meetup.zig
const std = @import("std");

pub const Month = enum {
    january,
    february,
    march,
    april,
    may,
    june,
    july,
    august,
    september,
    october,
    november,
    december,
};

pub const Week = enum {
    first,
    second,
    third,
    fourth,
    teenth,
    last,
};

pub const DayOfWeek = enum {
    monday,
    tuesday,
    wednesday,
    thursday,
    friday,
    saturday,
    sunday,
};

pub fn meetup(year: u12, month: Month, week: Week, day_of_week: DayOfWeek) [10]u8 {
    _ = year;
    _ = month;
    _ = week;
    _ = day_of_week;
    @compileError("please implement the meetup function");
}

File: generators/exercises/meetup.py
from lib import zstr

HEADER = (
    "const Month = meetup.Month;\n"
    "const Week = meetup.Week;\n"
    "const DayOfWeek = meetup.DayOfWeek;\n"
)

_MONTHS = [
    "january",
    "february",
    "march",
    "april",
    "may",
    "june",
    "july",
    "august",
    "september",
    "october",
    "november",
    "december",
]


def gen_case(case):
    inp = case["input"]
    year = inp["year"]
    month = _MONTHS[inp["month"] - 1]
    week = inp["week"].lower()
    day = inp["dayofweek"].lower()
    expected = case["expected"]
    return (
        f"    const expected = {zstr(expected)};\n"
        f"    const actual = meetup.meetup({year}, .{month}, .{week}, .{day});\n"
        f"    try testing.expectEqualStrings(expected, &actual);\n"
    )
